cylc_singleuser.py

```diff
@@ -119,6 +160,14 @@ def start(self):
         signal.signal(signal.SIGINT, app.signal_handler)
         app.listen(self._port)
         ioloop.PeriodicCallback(app.try_exit, 100).start()
+        # Discover workflows on intiial start up.
+        ioloop.IOLoop.current().add_callback(self.ws_mgr.gather_workflows)
+        # Arbitrary intervals (msec) chosen for the two periodic callbacks
+        # below. These will likely change or be removed with PUB/SUB updates.
+        # Check for new workflows every 19sec.
+        ioloop.PeriodicCallback(self.ws_mgr.gather_workflows, 19000).start()
+        ioloop.PeriodicCallback(
+            self.data_mgr.entire_workflow_update, 5000).start()
```

Comment:
These two `PeriodiCallback`'s are appearing with a warning on my IDE. Checking the signature of `PeriodicCallback` in Tornado source code it specifies the type as `Callable[[], None]`.

From [the typing docs about Callable](https://docs.python.org/3/library/typing.html#callable) 

>Frameworks expecting callback functions of specific signatures might be type hinted using Callable[[Arg1Type, Arg2Type], ReturnType].

So the Tornado function signature says `Callable[[], None]` which is equivalent to a `Callable` with no parameters, that does not return anything.

`async` methods are marked as [`Awaitable`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Awaitable)s.

Still trying to confirm that these are being executed correctly every 19 and 5 secs.

Reply:
I do some one initial start up, and yes still done every 19 and 5 secs respectively data manager one will likely change/be-removed with incremental updates.

Reply:
Ok, did some testing, and the function is called, but I believe it does not need to be `async`. It is not resolved with `await`, or executed in the main loop. It is [simply returned](https://github.com/tornadoweb/tornado/blob/f65b1802a549bab73b4a6721bbfc47f82954d156/tornado/ioloop.py#L907).